http-server/src/main/java/io/micronaut/http/server/util/locale/HttpLocalizedMessageSource.java

```
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.server.util.locale;
+
+import io.micronaut.context.AbstractLocalizedMessageSource;
+import io.micronaut.context.MessageSource;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.util.LocaleResolver;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.runtime.http.scope.RequestAware;
+import io.micronaut.runtime.http.scope.RequestScope;
+import java.util.Locale;
+
+/**
+ * A {@link RequestScope} which uses the current {@link HttpRequest} to resolve the locale and hence return the localized messages.
+ * @author Sergio del Amo
+ * @since 3.4.x
+ */
+@RequestScope
+public class HttpLocalizedMessageSource extends AbstractLocalizedMessageSource<HttpRequest<?>> implements RequestAware {
+    private Locale locale;
+    /**
+     * @param localeResolver The locale resolver
+     * @param messageSource  The message source
+     */
+    public HttpLocalizedMessageSource(LocaleResolver<HttpRequest<?>> localeResolver, MessageSource messageSource) {
+        super(localeResolver, messageSource);
+    }
+
+    @Override
+    @NonNull
+    protected Locale getLocale() {
+        return locale;
```

The return value should never be `null` but the code as written it could be, probably need to be able to return a default locale if non is available

---

I injected `HttpLocaleResolutionConfiguration` to set a default locale in the constructor.

 As I understood by being `RequestAware` the `setRequest` method would be called when the object was created and hence the locale was set. 

But I think now it is safer

---

We know the request will be set so I don't see the nullability as a concern. If it is null you could throw an exception as that should never happen and keep the contract as NonNull

---

I kept the contract as `@NonNull`. What I have done is set in the constructor as well.

---

Right but that is wrong imo. We can't assume the default is coming from that configuration. We can only rely on the contract of the interface

---

What exception do you want me to throw if request is null?

---

IllegalStateException

---

fixed via https://github.com/micronaut-projects/micronaut-core/pull/6870/commits/a7359f01a21447439d4d1469b9466ad2e603a290